nat/traversal/nat_proxy.go

```diff
@@ -41,6 +41,11 @@ type NATProxy struct {
 
 func (np *NATProxy) consumerHandOff(consumerPort int, remoteConn *net.UDPConn) chan struct{} {
 	stop := make(chan struct{})
+	if np.socketProtect == nil {
+		// shutdown pinger session since openvpn client will connect directly (without NATProxy)
```

Comment:
Why session is started at all, if you need to shut it down e.g. DI should launch noopSession

Reply:
pinger session is needed and at this point it already did its work, that why we shutting it down.